Agraus/c make revision

AndrewGraus opened the pull request:
Here's the full Cmake Revisison. I've tested it on biomes and lawrencium, so it should work on both. To compile it you can simply run the build_EcoSIM.sh shell script. Additionally you can go into the shell script and set options like debug or specify compilers (otherwise it will just use whatever compilers the environment variables point to).

On biomes the modules I'm using are:
gcc/11.4.0
python/3.11.6-gcc-11.4.0
cmake/3.27.7

It compiles and runs in coupled mode as well, and I've configured it to compile along with ATS. This pull also contains some minor changes to the coupler and SurfPhysMod for the snow implementation.

jinyun1tang reviewed (commented):
I looked through the build_EcoSIM.sh script, some changes can be made to make it more flexible. Please see my recommendations.

AndrewGraus commented on build_EcoSIM.sh:
```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
```
I'm basing it off the scripts to install ATS, it should work using either ./build_EcoSIM.sh or sh build_EcoSIM.sh, From looking it up it seems like sh is faster than bash in some cases, I assume that matters more for a more complex build script like ATS has.

AndrewGraus commented on build_EcoSIM.sh:
```diff
@@ -0,0 +1,164 @@
+#Leave empty to just use the environment variable compiler
+CC="/opt/homebrew/bin/gcc-12"
+CXX="/opt/homebrew/bin/g++-12"
+FC="/opt/homebrew/bin/gfortran-12"
```
Working on this now.

jinyun1tang replied:
can you also add a test option from the argument to do regression test if it is passed in? Sometimes, people just want to rebuild the code, not doing the test.

AndrewGraus replied:
I've added in command line options for the compilers, you can also turn off regression testing by going into the build script and turning off the regression tests option (regression_test=0). I am thinking of turning all of those flags in the build script into command line options as well, do you think it is worth spending time on?

smolins commented:
Instructions should just say

./build_EcoSIM.sh

I wonder whether everything in build_EcoSIM.sh is bash shell format,
though.




On Thu, Aug 29, 2024 at 4:56 PM Andrew Graus ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In build_EcoSIM.sh
> <https://github.com/jinyun1tang/EcoSIM/pull/58#discussion_r1737513193>:
>
> > @@ -0,0 +1,164 @@
> +#!/bin/bash
>
> I'm basing it off the scripts to install ATS, it should work using either
> ./build_EcoSIM.sh or sh build_EcoSIM.sh, From looking it up it seems like
> sh is faster than bash in some cases, I assume that matters more for a more
> complex build script like ATS has.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jinyun1tang/EcoSIM/pull/58#discussion_r1737513193>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADXOTGVAXUQHR525BA3F6W3ZT6YKHAVCNFSM6AAAAABM5AIZSWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDENZQGY2DCMJUGA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

smolins commented:
Why is  travis = 0 there?  This seems pretty old stuff from the old CI.



On Thu, Aug 29, 2024 at 5:14 PM Sergi Molins Rafa ***@***.***> wrote:

> Instructions should just say
>
> ./build_EcoSIM.sh
>
> I wonder whether everything in build_EcoSIM.sh is bash shell format,
> though.
>
>
>
>
> On Thu, Aug 29, 2024 at 4:56 PM Andrew Graus ***@***.***>
> wrote:
>
>> ***@***.**** commented on this pull request.
>> ------------------------------
>>
>> In build_EcoSIM.sh
>> <https://github.com/jinyun1tang/EcoSIM/pull/58#discussion_r1737513193>:
>>
>> > @@ -0,0 +1,164 @@
>> +#!/bin/bash
>>
>> I'm basing it off the scripts to install ATS, it should work using either
>> ./build_EcoSIM.sh or sh build_EcoSIM.sh, From looking it up it seems like
>> sh is faster than bash in some cases, I assume that matters more for a more
>> complex build script like ATS has.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/jinyun1tang/EcoSIM/pull/58#discussion_r1737513193>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ADXOTGVAXUQHR525BA3F6W3ZT6YKHAVCNFSM6AAAAABM5AIZSWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDENZQGY2DCMJUGA>
>> .
>> You are receiving this because you are subscribed to this thread.Message
>> ID: ***@***.***>
>>
>

AndrewGraus commented:
I've tested it with both and running it with bash versus sh doesn't cause
any issues on my mac at least.

As for the Travis stuff, it was in the old build Makefile and I just left
it in, in case it was needed. I can remove it if we don't need it.

-Andrew

On Thu, Aug 29, 2024 at 5:16 PM Sergi Molins ***@***.***>
wrote:

> Why is travis = 0 there? This seems pretty old stuff from the old CI.
>
>
>
> On Thu, Aug 29, 2024 at 5:14 PM Sergi Molins Rafa ***@***.***> wrote:
>
> > Instructions should just say
> >
> > ./build_EcoSIM.sh
> >
> > I wonder whether everything in build_EcoSIM.sh is bash shell format,
> > though.
> >
> >
> >
> >
> > On Thu, Aug 29, 2024 at 4:56 PM Andrew Graus ***@***.***>
> > wrote:
> >
> >> ***@***.**** commented on this pull request.
> >> ------------------------------
> >>
> >> In build_EcoSIM.sh
> >> <https://github.com/jinyun1tang/EcoSIM/pull/58#discussion_r1737513193>:
>
> >>
> >> > @@ -0,0 +1,164 @@
> >> +#!/bin/bash
> >>
> >> I'm basing it off the scripts to install ATS, it should work using
> either
> >> ./build_EcoSIM.sh or sh build_EcoSIM.sh, From looking it up it seems
> like
> >> sh is faster than bash in some cases, I assume that matters more for a
> more
> >> complex build script like ATS has.
> >>
> >> —
> >> Reply to this email directly, view it on GitHub
> >> <https://github.com/jinyun1tang/EcoSIM/pull/58#discussion_r1737513193>,
>
> >> or unsubscribe
> >> <
> https://github.com/notifications/unsubscribe-auth/ADXOTGVAXUQHR525BA3F6W3ZT6YKHAVCNFSM6AAAAABM5AIZSWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDENZQGY2DCMJUGA>
>
> >> .
> >> You are receiving this because you are subscribed to this
> thread.Message
> >> ID: ***@***.***>
> >>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jinyun1tang/EcoSIM/pull/58#issuecomment-2319521802>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD756HYJLUEBTFD4VOW3EJLZT62WNAVCNFSM6AAAAABM5AIZSWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMJZGUZDCOBQGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

jinyuntang commented:
for things other than building the code, I think it is better to turn it
off by default, and allow users to turn them on by passing from the
argument.

On Fri, Aug 30, 2024 at 9:31 AM Andrew Graus ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In build_EcoSIM.sh
> <https://github.com/jinyun1tang/EcoSIM/pull/58#discussion_r1739091231>:
>
> > +mpi=0
> +shared=0
> +verbose=0
> +sanitize=0
> +travis=0
> +regression_test=1
> +
> +precision="double"
> +prefix=""
> +systype=""
> +
> +
> +#Leave empty to just use the environment variable compiler
> +CC="/opt/homebrew/bin/gcc-12"
> +CXX="/opt/homebrew/bin/g++-12"
> +FC="/opt/homebrew/bin/gfortran-12"
>
> I've added in command line options for the compilers, you can also turn
> off regression testing by going into the build script and turning off the
> regression tests option (regression_test=0). I am thinking of turning all
> of those flags in the build script into command line options as well, do
> you think it is worth spending time on?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jinyun1tang/EcoSIM/pull/58#discussion_r1739091231>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACTQV3QHUZU6UONUAO47RPDZUCM7LAVCNFSM6AAAAABM5AIZSWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDENZTGAYDSNBVHE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Jinyun (He/him)
Staff Scientist
Lawrence Berkeley National Laboratory
1 Cyclotron Rd., MS 74R316C
Berkeley, CA 94720
tel: 510 486-5792, fax: 510 486-7070

AndrewGraus commented:
I've added in all the options as command line options and updated the README

smolins commented:
Functionally this seems fine to me.
We may want to consider nicer formatting for the options. For example,
prepending double dashes in the options, e.g. "--verbose=0" rather than
"verbose=0". And adding a "--help" option to list all possible
options without running the script.

On Fri, Aug 30, 2024 at 10:50 AM Andrew Graus ***@***.***>
wrote:

> I've added in all the options as command line options and updated the
> README
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jinyun1tang/EcoSIM/pull/58#issuecomment-2322058792>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADXOTGTGHYZXOJD357KBVBTZUCWETAVCNFSM6AAAAABM5AIZSWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMRSGA2TQNZZGI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

jinyuntang commented:
Yes, I do think adding double dash "--" and "--help" is helpful.

On Fri, Aug 30, 2024 at 11:13 AM Sergi Molins ***@***.***>
wrote:

> Functionally this seems fine to me.
> We may want to consider nicer formatting for the options. For example,
> prepending double dashes in the options, e.g. "--verbose=0" rather than
> "verbose=0". And adding a "--help" option to list all possible
> options without running the script.
>
> On Fri, Aug 30, 2024 at 10:50 AM Andrew Graus ***@***.***>
> wrote:
>
> > I've added in all the options as command line options and updated the
> > README
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/jinyun1tang/EcoSIM/pull/58#issuecomment-2322058792>,
>
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ADXOTGTGHYZXOJD357KBVBTZUCWETAVCNFSM6AAAAABM5AIZSWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMRSGA2TQNZZGI>
>
> > .
> > You are receiving this because you commented.Message ID:
> > ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jinyun1tang/EcoSIM/pull/58#issuecomment-2322091725>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACTQV3VZ4MZNGVW35A4JPALZUCY3PAVCNFSM6AAAAABM5AIZSWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMRSGA4TCNZSGU>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Jinyun (He/him)
Staff Scientist
Lawrence Berkeley National Laboratory
1 Cyclotron Rd., MS 74R316C
Berkeley, CA 94720
tel: 510 486-5792, fax: 510 486-7070

AndrewGraus commented:
Good ideas, I've added the help option and also changed all the options besides the compilers so that you don't have to define the booleans. For example now you just do --debug to turn on debug mode, otherwise it's off.
